.github/workflows/module_release.yml

```
@@ -24,13 +29,7 @@ jobs:
       - name: "Get version"
         id: "get_version"
         run: |
-          version=$(jq --raw-output '.version | gsub("[^0-9.]"; "")' metadata.json)
-          if [ -z "$version" ]; then
-            echo "Version is empty. Exiting workflow."
-            failure "Failed to retrieve version from metadata.json"
-          else
-            echo "version=$version" >> $GITHUB_OUTPUT
-          fi
+          ruby -e 'require "json"; version = JSON.parse(File.read("metadata.json"))["version"]; semver_regex = /\A(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?\z/; if version&.match?(semver_regex); puts "version=#{version}"; else; raise "Version #{version} is invalid. Exiting workflow."; end' >> $GITHUB_OUTPUT
```

```suggestion
          ruby <<-EOF >> $GITHUB_OUTPUT
            require "json"
            version = JSON.parse(File.read("metadata.json"))["version"]

            # from https://github.com/voxpupuli/metadata-json-lint/blob/b5e68049c6be58aa63263357bb0dcad8635a6829/lib/metadata-json-lint/schema.rb#L141-L150
            numeric = '(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)' # Major . Minor . Patch
            pre = '(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?' # Prerelease
            build = '(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?' # Build
            full = numeric + pre + build
            semver_regex = /\A#{full}\Z/
            if version&.match?(semver_regex)
              puts "version=#{version}"
            else
              raise "Version #{version} is invalid. Exiting workflow."
            end
          EOF
```

Having a one line command here is not maintainable. I added a suggestion to organize this so it is easier to review. I also added a comment to show where we are pulling the schema from. Please include that information in the commit message as well so future maintainers can find the source of that.